lighthouse-core/gather/driver.js

```diff
@@ -464,6 +467,25 @@ class Driver {
     }
   }
 
+  /**
+   * Evaluate an expression (optionally defined in a structured manner, see `createEvalCode`
+   * in `page-functions.js`).
+   * See `evaluateAsync`.
+   * @template T, R
+   * @param {string | ((...args: T[]) => R)} expressionOrMainFn
```

Comment:
I think this will have to be tweaked to get the args right (it should end up being a tuple). e.g. this would allow an array of strings to `fetcher`'s `injectIframe`, not a single string, and if the parameters are of mixed types I guess this would end up with an array of the union of the types?

Reply:
tried this but didnt work

![image](https://user-images.githubusercontent.com/4071474/98732346-a2cd5700-2364-11eb-8e38-25fa60410092.png)

![image](https://user-images.githubusercontent.com/4071474/98732379-acef5580-2364-11eb-812c-655c13f547f2.png)


then I found this: https://github.com/microsoft/TypeScript/issues/1024#issuecomment-343707528

Still parsing how this relates to variadic types / if this is now possible in typescript ...

Reply:
doh! just needed to change `args?: T[]` to `args?: T`